src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,166 @@
-import React from 'react';
+/* eslint-disable no-console */
+import React, { useCallback, useState } from 'react';
+import cn from 'classnames';
+import isEmail from 'validator/lib/isEmail';
+import { Comment } from '../types/Comment';
+import { postComment } from '../utils/api';
+import { Post } from '../types/Post';
+
+interface Props {
+  setComments: React.Dispatch<React.SetStateAction<Comment[] | null>>;
+  selectedPost: Post | null;
+}
+
+enum InputType {
+  NAME = 'name',
+  EMAIL = 'email',
+  BODY = 'body',
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  setComments,
+  selectedPost,
+}) => {
+  const [name, setName] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [body, setBody] = useState<string>('');
```

Comment:
I'd recommend you to use form fields as one state since this is one entity
```suggestion
  const [fields, setFields] = useState({
    name: '',
    email: '',
    body: ''
  });
```